### PR TITLE
Add PyTorch Tensor in CastType for post transforms

### PR DESCRIPTION
**Pull request by Nic-Ma**
### Description
This PR added PyTorch Tensor support for `CastType` transform, which can be used in post transforms.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh --codeformat --coverage`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: tests/test_cast_to_type.py [hunk 1]
```diff
@@ -10,15 +10,18 @@
...
 from monai.transforms import CastToType
 
 TEST_CASE_1 = [{"dtype": np.float64}, np.array([[0, 1], [1, 2]], dtype=np.float32), np.float64]
 
+TEST_CASE_2 = [{"dtype": torch.float64}, torch.tensor([[0, 1], [1, 2]], dtype=torch.float32), torch.float64]
+
 
 class TestCastToType(unittest.TestCase):
-    @parameterized.expand([TEST_CASE_1])
+    @parameterized.expand([TEST_CASE_1, TEST_CASE_1])
```
Comment (ericspod) [inline tests/test_cast_to_type.py:24]: `TEST_CASE_2` here?

Comment (Nic-Ma) [reply tests/test_cast_to_type.py]: Sorry for my typo, fixed in the latest commit.
Thanks.

**Comment by Nic-Ma** [comment]
Hi @ericspod ,

Thanks for your review, I updated the PR according to your comments.
Could you please help review again?

Thanks.

**Pull request merged**